nuxeo-runtime/nuxeo-stream/src/main/java/org/nuxeo/lib/stream/log/chronicle/ChronicleLogManager.java

```diff
@@ -0,0 +1,206 @@
+import org.nuxeo.lib.stream.log.LogLag;
+import org.nuxeo.lib.stream.log.LogPartition;
+import org.nuxeo.lib.stream.log.LogTailer;
+import org.nuxeo.lib.stream.log.RebalanceListener;
+import org.nuxeo.lib.stream.log.internals.AbstractLogManager;
+
+import java.io.Externalizable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.apache.commons.io.FileUtils.deleteDirectory;
+
+/**
+ * @since 9.3
+ */
+public class ChronicleLogManager extends AbstractLogManager {
+    private static final Log log = LogFactory.getLog(ChronicleLogManager.class);
+
+    /**
+     * Default retention duration for log
+     */
+    public static final String DEFAULT_RETENTION_DURATION = "4d";
+
+    protected final Path basePath;
+
+    protected final String retentionDuration;
+
+    public ChronicleLogManager(Path basePath) {
+        this.basePath = basePath;
+        this.retentionDuration = DEFAULT_RETENTION_DURATION;
+    }
+
+    /**
+     * Constructor
+     *
+     * @param basePath the base path.
+     * @param retentionDuration the retention duration. It is the time period the queue files will be retained. Once the
+     *            retention duration expires, the older files are candidates for being purged. The property can be
+     *            expressed as: 15s, 30m, 1h, 4d ... (where 's' is expressing a duration in seconds, 'm' in minutes,'h'
+     *            in hours and 'd' in days)
+     */
+    public ChronicleLogManager(Path basePath, String retentionDuration) {
+        this.basePath = basePath;
+        this.retentionDuration = retentionDuration == null ? DEFAULT_RETENTION_DURATION : retentionDuration;
+    }
+
+    public String getBasePath() {
+        return basePath.toAbsolutePath().toString();
+    }
+
+    @Override
+    public boolean exists(String name) {
+        File path = new File(basePath.toFile(), name);
+        //noinspection ConstantConditions
+        return path.isDirectory() && path.list().length > 0;
+    }
+
+    @Override
+    public void create(String name, int size) {
+        ChronicleLogAppender cq = ChronicleLogAppender.create(new File(basePath.toFile(), name), size, retentionDuration);
+        try {
+            cq.close();
+        } catch (Exception e) {
+            throw new RuntimeException("Cannot create and close " + name, e);
+        }
+    }
+
+    @Override
+    public boolean delete(String name) {
+        File path = new File(basePath.toFile(), name);
+        if (path.isDirectory()) {
+            deleteQueueBasePath(path);
+            return true;
+        }
+        return false;
+    }
+
+    protected LogLag getLagForPartition(String name, int partition, String group) {
+        long pos;
+        File path = new File(basePath.toFile(), name);
```

Comment:
You can replace it by `basePath.resolve(name)` as it, latter conversion to `String` is easier.